### ISSUE TITLE
посмотреть результаты для более честного молек. газа

### ISSUE THREAD

**Issue by Amarchuk** [LerkaKostuik/Course-work#12]
Еще одна короткая задача: вот в этой статье [THE CO-TO-H2 CONVERSION FACTOR AND DUST-TO-GAS RATIO ON KILOPARSEC SCALES IN NEARBY GALAXIES (2013)](https://arxiv.org/pdf/1212.1208.pdf) была сделана попытка оценить насколько фактор вычисления молек. газа CO-to-H2 отличается от среднего. NGC3627 там есть и кажется сильно весьма отличается от среднего общепринятого (поверхностный взгляд - среднее в 5 раз!). Надо посмотреть (хотя бы двухжидк.), насколько это скажется на результатах (предполагаю, что несильно). Если захочется - можно и грубую карту с Рис. 31 адаптировать

**Comment by ValerieKostiuk** [comment]
Я попробовала как-то адаптировать эту грубую карту. Посмотрите я этим занималась в I_CO-to-H2.ipynb. Карта двухжидкостной неустойчивости (как и предполагалось) не сильно меняется, хотя можно заметить, что на рукавах галактики появляется больше пикселей с неустойчивостью.

**Comment by Amarchuk** [comment]
Отличная работа! Карта конечно очень грубая и ее можно весьма вольно интерпретировать, но результаты хорошие. Мне сложно согласиться с 0.75 в рукавах (и в том, как вы отобразили по Sigma_HI - кажется слишком искусственным и трудно будет объяснить рецензенту), но в целом отлично. Три вещи которые здесь еще надо сделать:

* посмотреть как результат меняется для среднего alpha_CO по всему диску (=1.2, см Table 4)
* попробовать все же честно отобразить цвета на их шкалу и сделать по совмещенной карте (если не получится - я сделаю)
* смотреть также результаты для одножидкостного

**Comment by ValerieKostiuk** [comment]
Я попыталась построить карту честным образом. Если сравнивать карты для двухжидкостного критерия, то на первый взгляд они вообще не отличаются. Для одножидкостного изменения, конечно, легче заметить, но они, как мне кажется, не очень существенны. 
Я посмотрела, как будет выглядеть карта для метода галактического сдвига (galactic shear)(я не выставляла эти картинки), там карта получается вся в квадратиках (что не удивительно).
Может как-то попытаться "сгладить" эту карту? Или так с научной точки зрения нельзя делать? Я думаю можно было бы всё остальное пересчитать с новыми данными, хоть это может быть и не сильно изменило остальные карты. Но всё же как-то спокойнее оттого, что считаешь более честным способом.

**Comment by Amarchuk** [comment]
Отлично! Наиболее вероятно это потому, что все определяет по неустойчивости звезды, а не газ. Хорошо, что мы это проверили.
Я тут понял две вещи:

* мы забыли порисовать карты, где какая компонента определяющая (это легко сделать, для 628 я такие рисовал) - это важно и лучше сделать
* на картах для Q рисуйте пожалуйста в шкале, где видно какие области неуст. - например с 1 до 4 и чтоб были светлые неуст. Сейчас у нас граница непонятно где (цвет замывается). Как минимум все карты в `Results` надо перерисовать.

Насчет shear и перерисовки: это имеет смысл сделать только для alpha_CO=const, просто карта слишком уж крупная. Или как Вы раньше сделали - взять условно среднее на глаз (или обосновать) alpha_CO в центре и в спиралях - и как-то гладко его использовать.

Но в целом я думаю что надо сделать #11, пройтись Вам по остальным issue и закрыть, если сделано и доделать, если вдруг что-то мелкое осталось, просмотреть что все важное есть в результатах, сообщить мне, я посмотрю их и подумаю и мы потом созвонимся и их обсудим. Скорее всего надо публиковать то, что уже сделано и возможно в процессе написания статьи придет что-то еще на ум.

**Comment by ValerieKostiuk** [comment]
>мы забыли порисовать карты, где какая компонента определяющая (это легко сделать, для 628 я такие рисовал) - это важно и лучше сделать

Я правильно понимаю, что Вы имели в виду картинки, которые у Вас с статье в самом конце? Если да, то мне такое рисовать для всех параметров грав. нестабильности (Q_eff, shear, cold gas)?

**Comment by Amarchuk** [comment]
Рисовать только для Qeff, в остальных это же не имеет смысла?

> Я правильно понимаю, что Вы имели в виду картинки, которые у Вас с статье в самом конце?

Не совсем. Имеется в виду вот это:

Note that used approximation from Romeo & Falstad (2013) can also show which component is most unstable. In central parts of NGC 628, stars determine the dynamic status of a disc while molecular gas determines the instability for outer parts where Qg model works well. 

Т.е. можно взять их формулу (она простая) и сразу получать, какой компонент определяет неустойчивость. (в моих файлах можно посмотреть `n628_2D_small_v2.ipynb` -> 5.11 Romeo)

Или можно взять формулу для двухжидк. неустойчивости и посмотреть, какой из двух слагаемых больше(меньше). Возможно это даже интереснее и проще.

**Comment by ValerieKostiuk** [comment]
Я добавила новые картинки в Results.ipynb. И вроде бы всё, о чём Вы писали я сделала.

**Comment by Amarchuk** [comment]
Да, вроде все понятно и хорошо. Единственное в результатах я не понял что за карты для $\Sigma_{HI}>5 M_{\odot}/kpc^2$. Это вы совместили неуст. одним критерием и другим и так на них смотрите?

С этим разобраться и надо будет созвониться, сделать заготовку статьи, понять что все хорошо и собственно начать писать Вашу первую статью.

**Comment by ValerieKostiuk** [comment]
> Единственное в результатах я не понял что за карты для $\Sigma_{HI}>5 M_{\odot}/kpc^2$

Кажется я Вас до этого неправильно поняла, сейчас переделаю. Я не усвоила ключевую фразу, что $\Sigma_{HI}>5 M_{\odot}/kpc^2$ это критерий гравитационной неустойчивости.

**Comment by Amarchuk** [comment]
Да, только он не гравитационной, просто неустойчивости.

**Comment by ValerieKostiuk** [comment]
Я переделала и добавила результат.

**Comment by Amarchuk** [comment]
Там на карте опять сложно по цвету понять, где же неустойчивость, но в целом ладно. Видно, что F1 достаточно большая и согласие достаточно хорошее.

Я Вас попрошу сделать еще следующее в результатах (это упростит жизнь): собрать все варианты, что мы изучили, в табличку (где будет F1, recall, precision), всякие обрезанные и без нижнего квадрата тоже включать. И общую картинку со всеми-всеми картами (постарайтесь подобрать цвета так, чтоб было видно четко, где неуст. области, сами карты можно по убыванию F1 расположить). Вот после этого точно уже будем обсуждать.

**Comment by ValerieKostiuk** [comment]
Я добавила табличку и картинку с картами в Results.ipynb. Для того чтобы подобрать цвета, я на каждой карте рисовала чёрным цветом граничные области между устойчивыми и неустойчивыми областями (они немного видны на границах между светлой и тёмной областями).

**Comment by Amarchuk** [comment]
Замечательно! Пара замечаний:

- добавить Qgas (одножидкостная модель)
- указать для трех нижних, по полной ли они карте или по обрезанной
- меня немного постфактум смущает, что везде SFR разное - становится сложно методологически делать выводы. Стоит сделать следующее: добавить еще одну колонку, где будет только F-мера при SFR_lim=0.045. Соответственно для грав. неуст. результат не изменится, а для остальных - увидим.

**Comment by ValerieKostiuk** [comment]
Я обновила Results.ipynb и учла Ваши замечания.
> указать для трех нижних, по полной ли они карте или по обрезанной

А зачем это указывать, ведь у меня что для обрезанной, что для полной карты порог для SFR одинаков и по сути эти картинки ничем отличаться не будут? Хотя для полной карты можно понизить этот уровень, значения F-меры позволяют нам это сделать, но мне кажется это будет выглядеть уже немного неадекватно.

**Comment by Amarchuk** [comment]
> Я обновила Results.ipynb и учла Ваши замечания.

Отлично! И Вы даже добавили карты, что я не просил - но они оказались полезны (только Qgas добавьте в таблицу  пожалуйста).

И что посчитали последнюю колонку в таблице - тоже отлично, интересный результат конечно (я надеялся что станет получше в пользу грав. неуст., но похоже не особо). Если можно несложно посчитать и полноту с точностью - это было бы полезно.

**Карты:** их опять надо переделать - например опять для грав. не видно, где граница неустойчивости (Q=3, а там уже на шкале темно и не видно). Давайте чтобы раз и навсегда решить этот вопрос - сделайте как Вы для Ромео делали: просто два цвета - неуст/уст и сверху контуры звездообразования как сейчас. Никаких больше шкал серого, чтоб не приходилось в каждом случае подбирать цвета (еще пример, почему сейчас плохо - карта Sigma_HI > 5 - там же совсем не видно, где эта самая пятерка по цвету).

> А зачем это указывать, ведь у меня что для обрезанной, что для полной карты порог для SFR одинаков и по сути эти картинки ничем отличаться не будут?

Во-первых для единообразия, во-вторых - ведь статистики должны меняться. Т.е. да, порог мы берем тот же, но согласие у нас будет разное по качеству (для грав. моделей это ведь так, почему для остальных бы и нет?).

**Comment by ValerieKostiuk** [comment]
Я переделала карты. Посмотрите, пожалуйста, может мне ещё в них что-то нужно доделать.

**Comment by Amarchuk** [comment]
Отлично, теперь никаких вопросов они вызывать не будут. И то, что добавили данные в таблицу - тоже хорошо.

Добавьте пожалуйста еще одножидкостную модель (грав. неуст. только с газом) в таблицу, то сравнительно простое, сделайте #13 и хватит.
